org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/classpath/ResolveClassPathsHandler.java

```diff
@@ -42,11 +44,16 @@
    * Resolves class path for a java project.
    *
    * @param arguments a list contains project URI
+   * @param pm a progress monitor
    * @return the class paths entries
    */
-  public static List<String> resolveClasspaths(List<Object> arguments) {
+  public static List<String> resolveClasspaths(List<Object> arguments, IProgressMonitor pm) {
     String projectUri = (String) arguments.get(0);
 
+    if (pm.isCanceled()) {
```

Comment:
It's nice to check here, but it would be better to pass the pm into "getProjectClasspath" and to check inside the "for" loop.

Reply:
ok, done